core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -376,6 +379,21 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
+    private SessionProtocol sessionProtocol() {
```

Comment:
This method is called by every request. For micro-optimization and clarity, how about assigning `SessionProtocol` as a member field of this class?
The value could start with H1(C) and be updated to `H2C` when `UpgradeEvent` is received.

Reply:
Good idea! Addressed. 😉